ambry-store/src/main/java/com.github.ambry.store/StoreManager.java

```diff
@@ -80,9 +81,11 @@ public void start()
       if (!file.exists()) {
         throw new IllegalStateException("Mount path does not exist " + replica.getMountPath());
       }
+      // Partition id is used as the id of a store.
+      String storeId = replica.getPartitionId().toString();
```

Comment:
to note - Ming looked into `getBytes()` and it seems that the main implementation also embeds the version id. We decided to not use that as it might not be meaningful in the metric.
Instead, we decided to stick with `getString()`. However, since `getString()` is optional for an implementation, it cannot be relied on to return a unique store id. Therefore, we decided to make `getString()` mandatory by overriding it in the abstract class and defining that it should return a unique string for the partition.